xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/XWikiIcon.vue

```diff
@@ -30,6 +30,11 @@
     :class="iconDescriptor.cssClass"
   ></span>
 
+  <XWikiIcon
```

Comment:
Why does this component needs to be called recursively?

Reply:
It is to reuse the component logic for displaying `IMAGE` and `FONT` icons when the `REMOTE` icon descriptor as been fetched.
I used recursion because it required the least amount of changes, but actually yes there could be a way to do it without recursion.